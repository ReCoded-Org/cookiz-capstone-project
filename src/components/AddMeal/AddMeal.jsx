import { useTranslation } from "next-i18next";
import { useState } from "react";
import { HiPlusCircle } from "react-icons/hi";
import { IoAddCircleOutline, IoTrashBinOutline } from "react-icons/io5";

export default function AddMeal() {
    const [mealImage, setMealImage] = useState("../images/header.jpg");
    const [mealName, setMealName] = useState("");
    const [dietary, setDietary] = useState([]); // I know it seems unnecessary I put maybe we can use for filter
    const [vegan, isVegan] = useState(false);
    const [vegetarian, isVegetarian] = useState(false);
    const [glutenFree, isGlutenFree] = useState(false);
    const [ingredient, setIngredient] = useState(""); //it keeps state of place adjacent to plus icon
    const [ingredients, setIngredients] = useState([]); // it keeps array of all ingredients
    const [deliveryMin, setDeliveryMin] = useState("");
    const [price, setPrice] = useState("");

    function chooseImage(e) {
        setMealImage(URL.createObjectURL(e.target.files[0]));
    }
    function deleteIngredients(ingredientId) {
        setIngredients(
            ingredients.filter((item, itemId) => itemId !== ingredientId)
        );
    }
    function addIngredients(ingredient) {
        //add single ingredient to list
        setIngredients([...ingredients, ingredient]);
    }
    function checkVegan() {
        isVegan(!vegan);
        vegan ? setDietary([...dietary, "vegan"]) : null;
    }
    function checkVegetarian() {
        isVegetarian(!vegetarian);
        vegetarian ? setDietary([...dietary, "vegetarian"]) : null;
    }
    function checkGluten() {
        isGlutenFree(!glutenFree);
        glutenFree ? setDietary([...dietary, "gluten free"]) : null;
    }
    const { t } = useTranslation("addmeal");

    return (
        <div className='flex h-fit w-[340px] flex-col items-start justify-start gap-3 overflow-hidden'>
            <div
                style={{
                    backgroundImage: `url(${mealImage})`,
                    backgroundSize: "cover",

                    border: "none",
                }}
                className='mx-0 mt-0 flex h-48 w-full items-center justify-center overflow-hidden bg-center bg-no-repeat'
            >
                <div className='flex items-center justify-between gap-28 text-white md:gap-80 lg:gap-y-96'>
                    {/* <HiArrowSmLeft
                        className='cursor-pointer'
                        style={{ fontSize: "40px" }}
                    /> */}
                    <input
                        type='file'
                        id='add-image'
                        className='peer hidden'
                        required=''
                        onChange={chooseImage}
                    />
                    <label
                        htmlFor='add-image'
                        className='inline-flex w-full cursor-pointer items-center justify-between'
                    >
                        <div className='block'>
                            <HiPlusCircle
                                className='cursor-pointer'
                                style={{ fontSize: "40px" }}
                            />
                        </div>
                    </label>

                    {/* <HiArrowSmRight
                        className='cursor-pointer'
                        style={{ fontSize: "40px" }}
                    /> */}
                </div>
            </div>

            <input
                type='text'
                placeholder={t("fried-chips")}
                className='ml-5 w-2/3 rounded border-none bg-[#00494533] text-center text-sm focus:ring-1 focus:ring-[#004945] lg:text-base'
                required
                value={mealName}
                onChange={(e) => {
                    setMealName(e.target.value);
                }}
            />
            <p className='ml-3 text-sm lg:text-base'> {t("dietary")}</p>
            <ul className='ml-5 flex w-full items-center justify-start gap-3'>
                <li>
                    <input
                        type='checkbox'
                        id='react-option'
                        className='peer hidden'
                        required=''
                        onChange={() => checkVegetarian()}
                    />
                    <label
                        htmlFor='react-option'
                        className='inline-flex w-full cursor-pointer items-center justify-between rounded-full bg-[#00494533] p-1 ring-1 ring-[#00494533] peer-checked:ring-[#004945]'
                    >
                        <div className='block'>
                            <div className='w-full text-xs '>
                                {t("vegeterian")}
                            </div>
                        </div>
                    </label>
                </li>
                <li>
                    <input
                        type='checkbox'
                        id='flowbite-option'
                        className='peer hidden'
                        onChange={() => checkVegan()}
                    />
                    <label
                        htmlFor='flowbite-option'
                        className='inline-flex w-full cursor-pointer items-center justify-between rounded-full bg-[#00494533] p-1 ring-1 ring-[#00494533] peer-checked:ring-[#004945]'
                    >
                        <div className='block'>
                            <div className='w-full text-xs'>{t("vegan")}</div>
                        </div>
                    </label>
                </li>
                <li>
                    <input
                        type='checkbox'
                        id='angular-option'
                        className='peer hidden'
                        onChange={() => checkGluten()}
                    />
                    <label
                        htmlFor='angular-option'
                        className='inline-flex w-full cursor-pointer items-center justify-between rounded-full bg-[#00494533] p-1 ring-1 ring-[#00494533] peer-checked:ring-[#004945]'
                    >
                        <div className='block'>
                            <div className='w-full text-xs'>
                                {t("gluten-free")}
                            </div>
                        </div>
                    </label>
                </li>
            </ul>

            <p className='ml-3 text-sm lg:text-base'>{t("ingredients")}</p>
            <div className='ml-5 flex w-full flex-col items-start justify-start gap-3'>
                <div className='relative w-1/3'>
                    <input
                        type='text'
                        placeholder={t("tomatoes")}
                        value={ingredient}
                        className=' w-full rounded-full border-none bg-[#00494533] p-2 text-start text-xs focus:ring-1 focus:ring-[#004945] lg:text-sm'
                        onChange={(e) => setIngredient(e.target.value)}
                    ></input>
                    <IoAddCircleOutline
                        onClick={() => addIngredients(ingredient)}
                        className='absolute top-2 right-0.5 w-10 cursor-pointer'
                    ></IoAddCircleOutline>
                </div>

                <div className='flex w-full flex-wrap items-start justify-start gap-3'>
                    {ingredients.map((ingredient, id) => (
                        // eslint-disable-next-line react/jsx-key
                        <div className=' relative w-1/3'>
                            <p className='w-full rounded-full border-none bg-[#00494533] p-2 text-start text-xs text-zinc-800 lg:text-sm'>
                                {ingredient}
                            </p>
                            <IoTrashBinOutline
                                onClick={() => deleteIngredients(id)}
                                className='absolute top-2 right-0.5 w-10 cursor-pointer'
                            />
                        </div>
                    ))}
                </div>
            </div>
            <p className='ml-3 text-sm lg:text-base'>{t("delivery")}</p>
            <input
                type='text'
                value={deliveryMin}
                className='ml-5  rounded border-none text-xs ring-1 ring-[#004945] focus:rounded focus:ring-1 focus:ring-[#004945] lg:text-base'
                placeholder={t("time")}
                onChange={(e) => setDeliveryMin(e.target.value)}
            ></input>

            <p className='ml-3 text-sm lg:text-base'>{t("price")}</p>
            <input
                type='text'
                value={price}
                className='ml-5 mb-3 rounded border-none text-xs ring-1 ring-[#004945] focus:rounded focus:ring-1 focus:ring-[#004945] lg:text-base'
                placeholder={t("howmuch")}
                onChange={(e) => setPrice(e.target.value)}
            ></input>

            <button
                onClick={(e) => {
                    e.preventDefault();
                }}
                className=' absolute bottom-80 right-0  mr-2 rounded bg-[#004945] px-6 py-1 text-white'
            >
                {t("save")}
            </button>
        </div>
    );
}
