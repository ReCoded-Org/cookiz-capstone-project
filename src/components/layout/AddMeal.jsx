import { useEffect, useState } from "react";
import { HiArrowSmLeft, HiArrowSmRight, HiPlusCircle } from "react-icons/hi";
import { IoAddCircleOutline, IoTrashBinOutline } from "react-icons/io5";
export default function AddMeal() {
    const imageUrl = "../images/header.jpg";

    const [meal, setMeal] = useState({
        imageUrl: "",
        name: "",
        dietary: [],
        ingredients: ["tangerina", "cucumber"],
        minute: "",
        price: "",
    });

    useEffect(() => {}, [meal]);

    function addIngredient(value) {
        const obj = {
            ...meal,
            ingredients: [...meal["ingredients"], value],
        };
        setMeal(obj);
    }
    return (
        <div className='flex h-fit w-[340px] flex-col items-start justify-start gap-3 overflow-hidden'>
            <div
                style={{
                    backgroundImage: `url(${imageUrl})`,
                    backgroundSize: "cover",

                    border: "none",
                }}
                className='mx-0 mt-0 flex h-48 w-full items-center justify-center overflow-hidden bg-center bg-no-repeat'
            >
                <div className='flex items-center justify-between gap-28 text-white md:gap-80 lg:gap-y-96'>
                    <HiArrowSmLeft style={{ fontSize: "40px" }} />
                    <HiPlusCircle style={{ fontSize: "40px" }} />
                    <HiArrowSmRight style={{ fontSize: "40px" }} />
                </div>
            </div>

            <input
                type='text'
                placeholder='Fried Chips'
                className='ml-5 w-2/3 rounded border-none bg-[#00494533] text-center text-sm focus:ring-1 focus:ring-[#004945] lg:text-base'
                required
                value={Event.value}
                onChange={(e) => {
                    const obj = { ...meal, name: e.target.value };
                    setMeal(obj);
                }}
            />
            <p className='ml-3 text-sm lg:text-base'>Dietary</p>
            <div className='ml-5 flex w-full items-center justify-start gap-3'>
                <label
                    className=' w-1/4 rounded-full  bg-[#00494533] p-2 text-center text-xs text-zinc-800 lg:w-1/3 lg:text-sm'
                    htmlFor='vegetarianid'
                >
                    <input
                        type='checkbox'
                        id='vegetarianid'
                        value='vegetarianid'
                        checked=''
                        className='hidden'
                    />
                    Vegetarian
                </label>
                <label
                    className=' w-1/4 rounded-full  bg-[#00494533] p-2 text-center text-xs text-zinc-800 lg:w-1/3 lg:text-sm'
                    htmlFor='vegetarianid'
                >
                    <input
                        type='checkbox'
                        id='vegetarianid'
                        value='vegetarianid'
                        checked=''
                        className='hidden'
                    />
                    Vegan
                </label>
                <label
                    className=' w-1/4 rounded-full  bg-[#00494533] p-2 text-center text-xs text-zinc-800 lg:w-1/3 lg:text-sm'
                    htmlFor='vegetarianid'
                >
                    <input
                        type='checkbox'
                        id='vegetarianid'
                        value='vegetarianid'
                        checked=''
                        className='hidden'
                    />
                    Gluten Free
                </label>
            </div>
            <p className='ml-3 text-sm lg:text-base'>Ingredients</p>
            <div className='ml-5 flex w-full flex-col items-start justify-start gap-3'>
                <div className='relative w-1/3'>
                    <input
                        type='text'
                        placeholder='Tomatoes'
                        value={Event.value}
                        className=' w-full rounded-full border-none bg-[#00494533] p-2 text-start text-xs focus:ring-1 focus:ring-[#004945] lg:text-sm'
                    ></input>
                    <IoAddCircleOutline
                        onClick={(e) => {
                            addIngredient(e.target.value);
                        }}
                        className='absolute top-2 right-0.5 w-10'
                    ></IoAddCircleOutline>
                </div>

                <div className='flex w-full flex-wrap items-start justify-start gap-3'>
                    {meal["ingredients"].map((ingredient) => (
                        // eslint-disable-next-line react/jsx-key
                        <div className=' relative w-1/3'>
                            <p className='w-full rounded-full border-none bg-[#00494533] p-2 text-start text-xs text-zinc-800 lg:text-sm'>
                                {ingredient}
                            </p>
                            <IoTrashBinOutline className='absolute top-2 right-0.5 w-10' />
                        </div>
                    ))}
                </div>
            </div>
            <p className='ml-3 text-sm lg:text-base'>Delivery in</p>
            <input
                type='text'
                className='ml-5  rounded border-none text-xs ring-1 ring-[#004945] focus:rounded focus:ring-1 focus:ring-[#004945] lg:text-base'
                placeholder='30 min'
                value=''
            ></input>

            <p className='ml-3 text-sm lg:text-base'>Price</p>
            <input
                type='text'
                className='ml-5 mb-3 rounded border-none text-xs ring-1 ring-[#004945] focus:rounded focus:ring-1 focus:ring-[#004945] lg:text-base'
                placeholder='30 dollar'
                value=''
            ></input>

            <button className=' absolute bottom-72 right-0  rounded bg-[#004945] px-6 py-1 text-white'>
                Save
            </button>
        </div>
    );
}
