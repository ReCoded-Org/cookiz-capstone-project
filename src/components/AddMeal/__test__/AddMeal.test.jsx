import renderer from "react-test-renderer";

import AddMeal from "../AddMeal";

// eslint-disable-next-line no-undef
it("renders correctly", () => {
    const tree = renderer.create(<AddMeal />).toJSON();
    // eslint-disable-next-line no-undef
    expect(tree).toMatchSnapshot();
});
