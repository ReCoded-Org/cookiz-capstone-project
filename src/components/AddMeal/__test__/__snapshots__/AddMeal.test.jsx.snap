// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex h-fit w-[340px] flex-col items-start justify-start gap-3 overflow-hidden"
>
  <div
    className="mx-0 mt-0 flex h-48 w-full items-center justify-center overflow-hidden bg-center bg-no-repeat"
    style={
      Object {
        "backgroundImage": "url(../images/header.jpg)",
        "backgroundSize": "cover",
        "border": "none",
      }
    }
  >
    <div
      className="flex items-center justify-between gap-28 text-white md:gap-80 lg:gap-y-96"
    >
      <input
        className="peer hidden"
        id="add-image"
        onChange={[Function]}
        required=""
        type="file"
      />
      <label
        className="inline-flex w-full cursor-pointer items-center justify-between"
        htmlFor="add-image"
      >
        <div
          className="block"
        >
          <svg
            className="cursor-pointer"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
                "fontSize": "40px",
              }
            }
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M10 18a8 8 0 100-16 8 8 0 000 16zm1-11a1 1 0 10-2 0v2H7a1 1 0 100 2h2v2a1 1 0 102 0v-2h2a1 1 0 100-2h-2V7z"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </label>
    </div>
  </div>
  <input
    className="ml-5 w-2/3 rounded border-none bg-[#00494533] text-center text-sm focus:ring-1 focus:ring-[#004945] lg:text-base"
    onChange={[Function]}
    placeholder="fried-chips"
    required={true}
    type="text"
    value=""
  />
  <p
    className="ml-3 text-sm lg:text-base"
  >
     
    dietary
  </p>
  <ul
    className="ml-5 flex w-full items-center justify-start gap-3"
  >
    <li>
      <input
        className="peer hidden"
        id="react-option"
        onChange={[Function]}
        required=""
        type="checkbox"
      />
      <label
        className="inline-flex w-full cursor-pointer items-center justify-between rounded-full bg-[#00494533] p-1 ring-1 ring-[#00494533] peer-checked:ring-[#004945]"
        htmlFor="react-option"
      >
        <div
          className="block"
        >
          <div
            className="w-full text-xs "
          >
            vegeterian
          </div>
        </div>
      </label>
    </li>
    <li>
      <input
        className="peer hidden"
        id="flowbite-option"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="inline-flex w-full cursor-pointer items-center justify-between rounded-full bg-[#00494533] p-1 ring-1 ring-[#00494533] peer-checked:ring-[#004945]"
        htmlFor="flowbite-option"
      >
        <div
          className="block"
        >
          <div
            className="w-full text-xs"
          >
            vegan
          </div>
        </div>
      </label>
    </li>
    <li>
      <input
        className="peer hidden"
        id="angular-option"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="inline-flex w-full cursor-pointer items-center justify-between rounded-full bg-[#00494533] p-1 ring-1 ring-[#00494533] peer-checked:ring-[#004945]"
        htmlFor="angular-option"
      >
        <div
          className="block"
        >
          <div
            className="w-full text-xs"
          >
            gluten-free
          </div>
        </div>
      </label>
    </li>
  </ul>
  <p
    className="ml-3 text-sm lg:text-base"
  >
    ingredients
  </p>
  <div
    className="ml-5 flex w-full flex-col items-start justify-start gap-3"
  >
    <div
      className="relative w-1/3"
    >
      <input
        className=" w-full rounded-full border-none bg-[#00494533] p-2 text-start text-xs focus:ring-1 focus:ring-[#004945] lg:text-sm"
        onChange={[Function]}
        placeholder="tomatoes"
        type="text"
        value=""
      />
      <svg
        className="absolute top-2 right-0.5 w-10 cursor-pointer"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z"
          fill="none"
          strokeMiterlimit="10"
          strokeWidth="32"
        />
        <path
          d="M256 176v160m80-80H176"
          fill="none"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="32"
        />
      </svg>
    </div>
    <div
      className="flex w-full flex-wrap items-start justify-start gap-3"
    />
  </div>
  <p
    className="ml-3 text-sm lg:text-base"
  >
    delivery
  </p>
  <input
    className="ml-5 rounded border-none text-xs ring-1 ring-[#004945] focus:rounded focus:ring-1 focus:ring-[#004945] lg:text-base"
    onChange={[Function]}
    placeholder="time"
    type="text"
    value=""
  />
  <p
    className="ml-3 text-sm lg:text-base"
  >
    price
  </p>
  <input
    className="ml-5 mb-3 rounded border-none text-xs ring-1 ring-[#004945] focus:rounded focus:ring-1 focus:ring-[#004945] lg:text-base"
    onChange={[Function]}
    placeholder="howmuch"
    type="text"
    value=""
  />
  <button
    className=" absolute bottom-80 right-0 mr-2 rounded bg-[#004945] px-6 py-1 text-white"
    onClick={[Function]}
  >
    save
  </button>
</div>
`;
